### PR TITLE
Adding PTP functional tests

### PR DESCRIPTION
**Pull request by dorzheh**
# Description

Added PTP E2E tests

Fixes # (N/A)

## Type of change

Please select the appropriate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ v] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [v ] cd  <repo_name>/features and execute FEATURES=ptp make functests

**Test Configuration**:

- Versions: 4.3.0-0.nightly-2019-12-13-180405
- Hardware: baremetal

## Checklist

- [x] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not being merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

**Comment by yanirq** [comment]
@dorzheh  please note that you will have to add a CI job for that as described here https://github.com/openshift-kni/baremetal-deploy/blob/master/features/TESTING.md

**Comment by yanirq** [comment]
@dorzheh  can you please separate additional performance tests to another PR ?  Its easier to track and check.

File: features/functests/ptp/ptp.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+						Expect(ok).NotTo(BeTrue(),
+							fmt.Sprintf("Slave node %s shouldn't contain %s label", n.Name, ptpGrandmasterNodeLabel))
+					}
+
+					slaves++
+				}
+			}
+
+			By("Checking that all nodes are labeled")
+			Expect(len(nodes.Items)).To(BeNumerically("==", grands+slaves), "not nodes are labeled properly")
```
[the remark this answers is not in the document]
Comment (dorzheh) [reply features/functests/ptp/ptp.go]: @SchSeba , correct, Equal supposed to be faster than BeNumerically (though for this  test  its not so critical). Anyway,  Thanks for pointing out!

**Comment by fedepaol** [comment]
I have one more question: if I understood the code correctly, this test checks if the deployment happened correctly. 
What it does not, is to test if the ptp feature is working properly.
Do we have a way to test that @SchSeba @nikita133 ?

**Comment by dorzheh** [comment]
> I have one more question: if I understood the code correctly, this test checks if the deployment happened correctly.
> What it does not, is to test if the ptp feature is working properly.
> Do we have a way to test that @SchSeba @nikita133 ?

@fedepaol ,assume we need more tests covering at least the following verification: 1. PTP configuration according to the applied profiles  2. Time synchronization. Am i right?

**Comment by fedepaol** [comment]
> @fedepaol ,assume we need more tests covering at least the following verification: 1. PTP configuration according to the applied profiles 2. Time synchronization. Am i right?

Yes, I am not a ptp expert but the idea is to be able to test that it's working.

**Comment by fedepaol** [comment]
/lgtm
/approve

After this get merged, we are going to move the tests to the new https://github.com/openshift-kni/cnf-features-deploy repo

**Pull request merged**